h5py/_conv.pyx

```diff
@@ -951,7 +972,7 @@ cpdef int unregister_converters() except -1:
     H5Tunregister(H5T_PERS_HARD, "boolenum2b8", -1, -1, boolenum2b8)
     H5Tunregister(H5T_PERS_HARD, "b82boolenum", -1, -1, b82boolenum)
 
-    H5Tunregister(H5T_PERS_HARD, "uint82b8", -1, -1, uint82b8)
-    H5Tunregister(H5T_PERS_HARD, "b82uint8", -1, -1, b82uint8)
+    H5Tunregister(H5T_PERS_HARD, "uint2bitfield", -1, -1, uint2bitfield)
+    H5Tunregister(H5T_PERS_HARD, "bitfield2uint", -1, -1, bitfield2uint)
```

Comment:
Looking at the HDF5 source code, I think the name (the string here) needs to either match, or be a NULL pointer or an empty string to unregister all matches with the function:

https://github.com/HDFGroup/hdf5/blob/ce5f1ff1c502a1f341560254e4ed3b0dc65cfade/src/H5T.c#L2863-L2866

Reply:
My read is that it probably needs to be an uninitialized string - an empty string will still evaluate to boolean `true`. This assumes that passing `-1` for both `src` and `dst` will effectively leave the `H5T_t` object pointers uninitialized. If they _are_ initialized, they will evaluate to `true` and the function won't be the matching criteria - I've tried to investigate this but I'm finding the code pretty difficult to interpret.

This is the first time I've written any cython, so would just like to run this by you before I push anything:

```python
cdef char* empty_str
H5T_unregister(H5T_PERS_HARD, empty_str, -1, -1, uint2bitfield)
```
Does this look right to you?

Reply:
Is there a way to test that the way we're calling `H5Tunregister` is actually unregistering all of the conversion functions we intend to? If not, and I can't see any existing unit tests, perhaps it would be better to be explicit and just unregister the conversion functions for every name we have to be sure, since the name quality comparison precedes the `src` and `dst` initialisation. What are your thoughts @takluyver?

Reply:
Not as far as I know, last time I looked at adding tests around the type code, the information I could get from HDF5 was very limited.

Reply:
I think an empty string literal - `""` - should work, because it should be terminated by a null byte, and that's what the C code will look at (there are no strings in C, only pointers to a char, which may have some other chars after it). Or passing `NULL` should work as well (in the C code `name && *name && ...`, a null pointer should be false for `name`, and an empty string, a pointer to a null byte, should be false for `*name`).

Unfortunately, no, I don't think there's a great way to test these. The `unregister_converters` functions is only called when Python is exiting, as part of cleanup that's only sometimes important.

Reply:
Yep, you're right. I was only checking `name` when setting `const char* name = ""`. `*name` does indeed return false. Alright, I'll opt for an empty string literal then :)